src/timeline/js/directives/ruler.js

```diff
@@ -156,60 +157,68 @@ App.directive("tlRuler", function ($timeout) {
         }
       });
 
-      //watch the scale value so it will be able to draw the ruler after changes,
-      //otherwise the canvas is just reset to blank
-      scope.$watch("project.scale + markers.length + project.duration", function (val) {
-        if (val) {
-
-          $timeout(function () {
+      function drawTicks() {
+      // drawTicks : function () {
+        $timeout(function () {
```

Comment:
I wonder if the removal of $timeout will cause any issue. I "think" that was originally added to prevent the UI from freezing while drawing thousands upon thousands of tick marks on the Ruler. However, since we are only drawing the visible ruler now... I guess it makes sense that this is not needed anymore, lol.

Reply:
@JacksonRG 
We could always init a new scope variable in controllers.js:
`$scope.scrollLeft = 0;`

And extend the scope.watch() with a new custom scope variable:
`"project.scale + markers.length + project.duration + scrollLeft"`

And update it on scroll:
```
element.on("scroll", function () {
        scope.$apply(function () {
          scope.scrollLeft = element.scrollLeft();
        });
```

Reply:
This would also remove the need to refactor drawTicks() into it's own method. I'm just not sure if this would have any serious performance implications. I'm guessing if the only part of Angular that is watching/bound to $scope.scrollLeft is this one watch... and if that watch is super fast at drawing ticks... it might be a non-issue.

Reply:
That seems to work nicely in my tests

Reply:
Also, one final suggestion... perhaps make the "right side" calculation go off the screen, so we don't have a gap on the right side of the ruler...
![Screenshot from 2021-05-21 13-26-23](https://user-images.githubusercontent.com/5551883/119182153-2ebceb80-ba38-11eb-8187-f45653155497.png)

Reply:
Sounds good! Thanks for the tips. I'll push an update to this later tonight

Reply:
Fixed. Rounded up to go off the screen. And you were right. scope.scrollLeft was a simple solution to let ruler watch for changes.